From b41259dd7ab6582c409288e8e084f9e937cab8b6 Mon Sep 17 00:00:00 2001
From: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
Date: Fri, 10 Apr 2020 10:20:26 +0530
Subject: [PATCH 1324/1851] mtd: spi-nor: Fix page size calculation

commit 2527b868e883a537254b5faf7da6e8906a0aa7e5 from
https://github.com/Xilinx/linux-xlnx.git

Do not shift the page, as we are not updating this when we start supporting
the ubifs. there are corner cases in ubifs where the page size can be
unaligned, in this case the page size is wrongly calculated, which is
causing failures in ubi stress test. where as in normal mtd tests and
flashcp we will never see unaligned page sizes.

Signed-off-by: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index be85419f68bb..55e678edd000 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -5563,7 +5563,6 @@ int spi_nor_scan(struct spi_nor *nor, const char *name,
 				nor->shift = 1;
 				info->sector_size <<= nor->shift;
 				info->page_size <<= nor->shift;
-				nor->page_size = info->page_size;
 				mtd->size <<= nor->shift;
 				nor->isparallel = 1;
 				nor->isstacked = 0;
-- 
2.31.1

